component/otelcol/consumer.go

```diff
@@ -0,0 +1,19 @@
+package otelcol
+
+import (
+	otelconsumer "go.opentelemetry.io/collector/consumer"
+)
+
+// Consumer is a combined OpenTelemetry Collector consumer which can consume
+// any telemetry signal.
+type Consumer interface {
+	otelconsumer.Traces
+	otelconsumer.Metrics
+	otelconsumer.Logs
+}
```

Comment:
So the way that the OTel collector works here is that an implementation can choose to handle a signal with a consumer, or return `ErrDataTypeIsNotSupported`, right?

Reply:
Right, though ErrDataTypeIsNotSupport isn't really intended to be used by consumers; we're just reusing it here for convenience. OpenTelemetry would give you a warning at load time if you're trying to send traces to a component that doesn't support it, but since we're combining everything into one consumer we have emit the errors at runtime instead.